### PR TITLE
JBIDE-9442 - Add remote debugger launch when running server via rse

### PR DESCRIPTION
**Pull request by robstryker**
Fix bug where always using default port, never user-specified

### CODE REVIEW

File: as/plugins/org.jboss.ide.eclipse.as.core/jbosscore/org/jboss/ide/eclipse/as/core/server/launch/CommandLineLaunchConfigProperties.java [hunk 1]
```diff
@@ -38,6 +39,7 @@ public DefaultKeySet() {
 			SHUTDOWN_COMMAND = "org.jboss.ide.eclipse.as.core.launch.SHUTDOWN_COMMAND";//$NON-NLS-1$
 			DETECT_STARTUP_COMMAND = "org.jboss.ide.eclipse.as.core.launch.DETECT_STARTUP_COMMAND";//$NON-NLS-1$
 			DETECT_SHUTDOWN_COMMAND = "org.jboss.ide.eclipse.as.core.launch.DETECT_SHUTDOWN_COMMAND";//$NON-NLS-1$
+			DEBUG_STARTUP_COMMAND = null;  // not relevant
```
Comment (fbricon) [inline as/plugins/org.jboss.ide.eclipse.as.core/jbosscore/org/jboss/ide/eclipse/as/core/server/launch/CommandLineLaunchConfigProperties.java]: wat?

File: as/plugins/org.jboss.ide.eclipse.as.rse.core/src/org/jboss/ide/eclipse/as/rse/core/RSEJBoss7LaunchConfigurator.java [hunk 2]
```diff
@@ -153,24 +155,46 @@ protected String getArgsOverrideExposedManagement(IServer server, String preArgs
...
+		String programArguments = getLaunchProgramArgs();
+		String vmArguments = getLaunchVMArgs();
 		String jar = getJar(jbossServer.getServer());
-
-		String command = "java "
-				+ vmArguments
-				+ " -jar " + jar + " "
-				+ IJBossRuntimeConstants.SPACE + programArguments 
-				+ "&";
+		String command = "java " + vmArguments + " -jar " + jar + " " + IJBossRuntimeConstants.SPACE + programArguments + "&";
```
Comment (fbricon) [inline as/plugins/org.jboss.ide.eclipse.as.rse.core/src/org/jboss/ide/eclipse/as/rse/core/RSEJBoss7LaunchConfigurator.java]: `" " + IJBossRuntimeConstants.SPACE` that's 2 spaces right?

File: as/plugins/org.jboss.ide.eclipse.as.rse.core/src/org/jboss/ide/eclipse/as/rse/core/RSEJBoss7LaunchConfigurator.java [hunk 3]
```diff
@@ -153,24 +155,46 @@ protected String getArgsOverrideExposedManagement(IServer server, String preArgs
...
+		return command;
+	}
+	
+	protected String getLaunchDebugCommand(JBossServer jbossServer) throws CoreException {
+		String programArguments = getLaunchProgramArgs();
+		String vmArguments = getLaunchVMArgs();
+		String jar = getJar(jbossServer.getServer());
+		String debugPort = getDebugPort();
+		String debugArgs = "-Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=" + debugPort + " -Xdebug ";
+		String command = "java " + debugArgs + vmArguments + " -jar " + jar + " " + IJBossRuntimeConstants.SPACE + programArguments + "&";
```
Comment (fbricon) [inline as/plugins/org.jboss.ide.eclipse.as.rse.core/src/org/jboss/ide/eclipse/as/rse/core/RSEJBoss7LaunchConfigurator.java]: `" " + IJBossRuntimeConstants.SPACE`

File: as/plugins/org.jboss.ide.eclipse.as.rse.core/src/org/jboss/ide/eclipse/as/rse/core/StandardRSEJBossStartLaunchDelegate.java [hunk 4]
```diff
@@ -87,4 +169,58 @@ protected boolean isStarted(IServer server) {
...
+	
+	protected void connectDebugger(ILaunchConfiguration configuration, String mode, 
+			IControllableServerBehavior beh) throws CoreException {
+		attachRemoteDebugger(beh.getServer(), getDebugPort(beh.getServer()), new NullProgressMonitor());
+	}
+	
+
+	private ILaunch attachRemoteDebugger(IServer server, int localDebugPort, IProgressMonitor monitor) 
+			throws CoreException {
+		monitor.subTask("Attaching remote debugger");
```
Comment (fbricon) [inline as/plugins/org.jboss.ide.eclipse.as.rse.core/src/org/jboss/ide/eclipse/as/rse/core/StandardRSEJBossStartLaunchDelegate.java]: if localDebugPort < 1 it's invalid, should throw an exception

Comment (robstryker) [inline as/plugins/org.jboss.ide.eclipse.as.rse.core/src/org/jboss/ide/eclipse/as/rse/core/StandardRSEJBossStartLaunchDelegate.java]: User can't type a negative number into the UI. I've added another check that if it's < 1, to use the default value.

**Pull request merged**